### DISCUSSION TITLE
Easier way to pipe all incoming mails for an alias to a script?

### DISCUSSION THREAD

**Discussion by DerKlops** [docker-mailserver/docker-mailserver#3015] [Q&A] [answered]
Hello everyone!

I would like to pipe all incoming mails for a address directly to a Python script.
This script receives the mails via STDIN, parses the mail and then do some stuff based on the content of the mail.

After some tinkering I have found a working solution.
But, is there an easier way to achieve this piping?

What am I doing now?

1. Add a new alias to "/etc/aliases"
```
mailpipe: "/tmp/scripts/mailpipe.py"
```
2. Add an alias via cli
```
setup alias add mailpipe@example.com mailpipe@localhost
```
3. Now I see in the log, everything works as expected.

```
Jan 21 13:54:19 mail postfix/local[3237]: 3704942FFA: to=<mailpipe@localhost>, relay=local, delay=0.07, delays=0.02/0.02/0/0.03, dsn=2.0.0, status=sent (delivered to command:  /tmp/docker-mailserver-scripts/mailpipe.py)
```
My question now is,  is there a way to realize this direct with the setup command?

```
setup alias add mailpipe@example "| /tmp/scripts/mailpipe.py"
```
will result in bounced mails.

```
Jan 21 14:06:55 mail postfix/local[6463]: E335C42FFA: to=<| /tmp/scripts/mailpipe.py@mail.example.com>, relay=local, delay=0.04, delays=0.01/0.01/0/0.01, dsn=5.1.1, status=bounced (unknown user: "| /tmp/scripts/mailpipe.py")
```

Cheers
DerKlops

**Comment by casperklein** [comment]
> My question now is, is there a way to realize this direct with the setup command?

Currently that's not possible as you noticed. You can request this as a new feature, but it's not guaranteed that someone will pick up this task. You can of cause provide a PR yourself.

**Comment by polarathene** [comment, marked as answer]
- `setup alias ...` is for creating an alias to one or more recipient addresses (local or remote). Postfix does not allow for virtual aliases to use other destination like you added to `/etc/aliases` which is different in Postfix.
- `/etc/aliases` is intended for local system users (not handled by Dovecot). You can also direct mail to a process instead like you configured.

You can use `postfix-aliases.cf` config file we support to append your own lines to `/etc/aliases` (_[here is the startup script that does that](https://github.com/docker-mailserver/docker-mailserver/blob/bba72daedf4ecf03a80eea571160371191237863/target/scripts/helpers/aliases.sh#L48-L59)_). Then just create alias to `mailpipe@localhost` like you've done in step 2.

`postfix-aliases.cf` does not seem to be mentioned in the docs besides a small FAQ entry. It technically [should be listed here](https://docker-mailserver.github.io/docker-mailserver/latest/config/advanced/optional-config/).

You can probably also use virtual domain alias if you want to forward all mail to the script. No mail accounts for that domain, using `postfix-virtual` (_`setup alias ...` updates this file_) with `@example.com mailpipe@localhost`.

**Reply by Jarrodsz** [reply]
Interesting.

I used to setup thunderbird and let it run on a vps to handle the incoming mails then apply rules to them which consisted of auto reply if certain keywords would exist. Remove spam and add certain mails to spam if a user would repeatetly send the same mails or got an auto reply. A bit harsh but i cant afford spending hours on my support mailboxes for stubborn questions or complaints 🤣

It works fairly well.

Plan to setup thunderbird again to do this i just coudent find the energy to script this.

Downside you would need to be anle to let thunderbird 24/7 so it can apply the rules to any incoming mail. Still some things did not work like extracting email and call external script with email as identifier to do certain actions.

**Reply by DerKlops** [reply]
Thats it...
I just use the postfix-aliases.cf file now.
Works as expected, no bounces, etc...
Thank you.
